eval/hf_runner.py

```diff
@@ -29,6 +29,17 @@ def generate_prompt(prompt_file, question, db_name, public_data):
     return prompt
 
 
+def dynamic_num_beams(prompt: str, tokenizer) -> int:
+    tokens = len(tokenizer.encode(prompt))
+    print(tokens)
+    if tokens <= 1024:
```

Comment:
Thanks for implementing this!

Should we still let users set something like "max beams"? The idea here is to a) let them choose a lower numbers of beams if they want (say, for latency) and b) to let them choose a higher number of beams if they want (say, for higher accuracy – and if they have access to high memory GPUs)

I'm thinking of something like this
```python
def dynamic_num_beams(prompt: str, tokenizer, max_beams = 4) -> int:
    tokens = len(tokenizer.encode(prompt))
    print(tokens)
    if tokens <= 1024:
        return max_beams
    elif tokens <= 1536:
        return max_beams//2
    else:
        return max_beams//4
```